File: src/main/java/cc/suvankar/moneytrail/user/UserPrincipal.java
package cc.suvankar.moneytrail.user;

import java.util.Collection;
import java.util.List;
import java.util.UUID;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

public record UserPrincipal(String email, UUID userId) implements UserDetails {

  @Override
  public Collection<? extends GrantedAuthority> getAuthorities() {
    return List.of();
  }

  @Override
  public String getPassword() {
    throw new UnsupportedOperationException("Password not used in UserPrincipal");
  }

  @Override
  public String getUsername() {
    return email;
  }
}

File: src/main/kotlin/cc/suvankar/moneytrail/user/UserPrincipal.kt
package cc.suvankar.moneytrail.user

import java.util.UUID
import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.userdetails.UserDetails

class UserPrincipal(val email: String, val userId: UUID) : UserDetails {
  override fun getAuthorities(): Collection<GrantedAuthority?>? {
    return listOf()
  }

  override fun getPassword(): String? {
    throw UnsupportedOperationException("Password not used")
  }

  override fun getUsername(): String? {
    return email
  }
}
